src/actions/auto-merge.ts

```diff
@@ -64,14 +70,27 @@ export async function run(): Promise<void> {
       core.setOutput('commentID', resp.id);
     }
 
-    await client.pulls.merge({
-      owner,
-      repo,
-      pull_number: configInput.pullRequestNumber,
-      merge_method: configInput.strategy,
-    });
+    const branchName = pullRequest.head.ref;
+    const baseBranchName = pullRequest.base.ref;
+
+    if (baseBranchName !== 'master' && baseBranchName !== 'main') {
```

Comment:
Let's have a separate function `shouldMergePRBasedOnBaseBranch`

Reply:
I moved this logic to a separate function in github.ts (develop branch), so if there is okay, I can merge these two branches for a full view.